diff --git a/CMakeLists.txt b/CMakeLists.txt
index 35618f0..8bcefc8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -206,8 +206,11 @@ else()
   message(STATUS "Not using flex and/or bison, using the checked-in scanner and parser")
 endif()
 
-target_include_directories(libtailslide PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
-target_include_directories(libtailslide PUBLIC libtailslide)
+target_include_directories(libtailslide PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libtailslide>
+        $<INSTALL_INTERFACE:include>
+        $<INSTALL_INTERFACE:include/tailslide>)
 target_sources(libtailslide PRIVATE ${LSLMINI_SCANNER} ${LSLMINI_PARSER} ${LSLMINI_PARSER_HEADER})
 
 # Suppress warning about unused yynerrs variable in Bison-generated code
@@ -225,8 +228,10 @@ if (NOT WIN32 AND NOT TAILSLIDE_FUZZER_INSTRUMENTATION)
 endif()
 
 
-install(TARGETS libtailslide DESTINATION lib)
+install(TARGETS libtailslide EXPORT unofficial-tailslide-targets ARCHIVE DESTINATION lib)
 install(DIRECTORY libtailslide/ DESTINATION include/tailslide FILES_MATCHING PATTERN "*.hh")
+install(EXPORT unofficial-tailslide-targets NAMESPACE unofficial::tailslide:: DESTINATION share/unofficial-tailslide)
+install(FILES cmake/unofficial-tailslide-config.cmake DESTINATION share/unofficial-tailslide)
 
 
 if (TAILSLIDE_BUILD_CLI)
diff --git a/cmake/unofficial-tailslide-config.cmake b/cmake/unofficial-tailslide-config.cmake
new file mode 100644
index 0000000..6887ff4
--- /dev/null
+++ b/cmake/unofficial-tailslide-config.cmake
@@ -0,0 +1 @@
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-tailslide-targets.cmake")
